source/Engine.cpp

```diff
@@ -1633,6 +1639,36 @@ void Engine::MoveShip(const shared_ptr<Ship> &ship)
 
 
 
+// Changing the players flagship during flight requires an update of all active
+// data that references the players flagship.
+void Engine::ChangePlayerFlagship(const shared_ptr<Ship> oldFlagship, shared_ptr<Ship> &newFlagship)
+{
+	// Remove active data in the old flagship.
+	if(oldFlagship)
+	{
+		oldFlagship->SetTargetAsteroid(nullptr);
+		oldFlagship->SetTargetFlotsam(nullptr);
+		oldFlagship->SetTargetShip(nullptr);
+		oldFlagship->SetTargetStellar(nullptr);
+		oldFlagship->SetTargetSystem(nullptr);
+	}
+	
+	// Let PlayerInfo handle the default administration (like the players escorts).
+	player.SetFlagship(newFlagship);
+	
+	// Update all ships that still have the old flagship as parent, for example
+	// NPCs that are following the player.
+	for(std::shared_ptr<Ship>& ship : ships)
```

Comment:
```suggestion
	for(const auto& ship : ships)
```

Reply:
The ship can have a SetParent() called on it, so suggesting `const` here seems a little bit inappropriate (even if the reference is const and not the ship itself).

Reply:
I changed it to `auto &`.